pyntcloud/io/bin.py

```diff
@@ -0,0 +1,80 @@
+#       HAKUNA MATATA
+
+# Contributed by: Nicholas Mitchell
+
+import numpy as np
+import pandas as pd
+
+
+def read_bin(filename, shape=None, **kwargs):
+    """ Read a _raw binary_ file and store all possible elements in pandas DataFrame.
+
+    If the shape of the array is known, it can be specified using
+    `shape`. The first three columns are used for x, y and z.
+    Otherwise the binary file is assumed have row-major format and
+    three columns are formed.
+
+    NOTE: binary files that are not `raw` will not behave as expected. If they
+    contain a header/footer with meta data, or were generated e.g. via Protobuf,
+    then bahviour is also undefined.
+
+    Parameters
+    ----------
+    filename: str
+        Path to the filename
+    shape: shape of array if known, optional.
+    **kwargs:
+    kwargs: numpy.fromfile supported kwargs
+        Check NumPy documentation for all possibilities.
+
+    Returns
+    -------
+    data: dict
+        If possible, elements as pandas DataFrames else a NumPy ndarray
+    """
+    data = {}
+
+    kwargs['dtype'] = kwargs.get('dtype', np.float32)
+    arr = np.fromfile(filename, **kwargs)
+    n_elements = arr.size
+
+    if shape is not None:
```

Comment:
I'm not really understanding what is the purpose of the shape argument or what are the possible values. Although it might be because of I'm not familiar with this kind of files, could you please elaborate a little bit more on the docstring on the parameter??

Reply:
I have added further information. Also have a look at the docstring of the function as well as the argument's docstring - there is some more information there. In general, the binary data could contain anything... it is a sequential stream of bytes. The shape of the array that will be loaded is the only real information that a user can specify (in addition to any kwargs, which can be passed to `numpy.fromfile()`)